packages/client/lib/OpenID4VCIClient.ts

```diff
@@ -1,72 +1,84 @@
 import {
+  AccessTokenRequestOpts,
   AccessTokenResponse,
   Alg,
   AuthorizationRequest,
   AuthorizationRequestOpts,
   AuthzFlowType,
-  CredentialMetadata,
+  CredentialMetadataV1_09,
   CredentialResponse,
   CredentialsSupported,
   EndpointMetadata,
   IssuanceInitiationWithBaseUrl,
+  OIDCVCIVersion,
   ProofOfPossessionCallbacks,
   ResponseType,
 } from '@sphereon/openid4vci-common';
-import { CredentialFormat } from '@sphereon/ssi-types';
+import {CredentialFormat} from '@sphereon/ssi-types';
 import Debug from 'debug';
 
-import { AccessTokenClient } from './AccessTokenClient';
-import { CredentialRequestClientBuilder } from './CredentialRequestClientBuilder';
-import { IssuanceInitiation } from './IssuanceInitiation';
-import { MetadataClient } from './MetadataClient';
-import { ProofOfPossessionBuilder } from './ProofOfPossessionBuilder';
-import { convertJsonToURI } from './functions';
+import {AccessTokenClient} from './AccessTokenClient';
+import {CredentialOfferIssuance, CredentialOfferStrategy, discoverOIDCVCIVersion, getStrategy, IssuanceInitiation} from './CredentialOffer';
+import {CredentialRequestClientBuilder} from "./CredentialRequestClientBuilder";
+import {ProofOfPossessionBuilder} from './ProofOfPossessionBuilder';
+import {convertJsonToURI} from './functions';
 
 const debug = Debug('sphereon:openid4vci:flow');
 
 export class OpenID4VCIClient {
   private readonly _flowType: AuthzFlowType;
-  private readonly _initiation: IssuanceInitiationWithBaseUrl;
   private _clientId?: string;
   private _kid: string | undefined;
   private _alg: Alg | string | undefined;
   private _serverMetadata: EndpointMetadata | undefined;
   private _accessTokenResponse: AccessTokenResponse | undefined;
-
-  private constructor(initiation: IssuanceInitiationWithBaseUrl, flowType: AuthzFlowType, kid?: string, alg?: Alg | string, clientId?: string) {
+  private readonly _oidcvciVersion: OIDCVCIVersion;
+  private readonly _strategy: CredentialOfferStrategy;
```

Comment:
rename strategy, as this is a high-level client which is delegating multiple responsibilities, we need to be more clear in the naming

Reply:
Changed to a more specific name.
**CredentialIssuanceClient**